manager-bundle/src/Command/MaintenanceModeCommand.php

```diff
@@ -42,27 +44,55 @@ public function __construct(string $maintenanceFilePath, Environment $twig, File
     protected function configure(): void
     {
         $this
-            ->setName('contao:maintenance-mode')
-            ->addArgument('state', InputArgument::REQUIRED, 'Use "enable" to enable and "disable" to disable the maintenance mode. If the state is already the desired one, nothing happens. You can also use "on" and "off".')
+            ->addArgument('state', InputArgument::OPTIONAL, 'Use "enable" to enable and "disable" to disable the maintenance mode. If the state is already the desired one, nothing happens. You can also use "on" and "off".')
             ->addOption('template', 't', InputOption::VALUE_REQUIRED, 'Allows to take a different Twig template name when enabling the maintenance mode.', '@ContaoCore/Error/service_unavailable.html.twig')
             ->addOption('templateVars', null, InputOption::VALUE_OPTIONAL, 'Add custom template variables to the Twig template when enabling the maintenance mode (provide as JSON).', '{}')
+            ->addOption('format', null, InputOption::VALUE_REQUIRED, 'The output format (txt, json)', 'txt')
             ->setDescription('Changes the state of the system maintenance mode.')
         ;
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $shouldEnable = \in_array($input->getArgument('state'), ['enable', 'on'], true);
+        $state = $input->getArgument('state');
         $io = new SymfonyStyle($input, $output);
 
-        if ($shouldEnable) {
+        if (\in_array($state, ['enable', 'on'], true)) {
             $this->enable($input->getOption('template'), $input->getOption('templateVars'));
             $io->success('Maintenance mode enabled.');
-        } else {
+
+            return 0;
+        }
+
+        if (\in_array($state, ['disable', 'off'], true)) {
             $this->disable();
             $io->success('Maintenance mode disabled.');
+
+            return 0;
         }
 
+        $isEnabled = $this->filesystem->exists($this->maintenanceFilePath);
+
+        if ('json' === $input->getOption('format')) {
+            $output->writeln(json_encode(
+                [
+                    'enabled' => $isEnabled,
+                    'maintenanceFilePath' => $this->maintenanceFilePath,
+                ],
+                JSON_THROW_ON_ERROR
+            ));
+
+            return 0;
+        }
+
+        if ($isEnabled) {
+            $io->note('Maintenance mode is enabled.');
```

Comment:
Why is this a note and not an info like in line 94? And why do you return an error when there is nothing wrong?

Reply:
in my perspective, the enabled maintenance mode should be some sort of warning in the console. That includes a different message design and an exit code. Same as the back end previously added a "warning" bell that your system was in maintenance.

Reply:
But `return 1` means "error", which means that a potential script will exit even though nothing is wrong.

Reply:
Also makes the exit code dependent on the `format` which seams weird.

Reply:
well what's the exit code for "this is enabled" then? :joy

Reply:
If you return anything other than zero, you cannot use this as part of your deployments, since your deployment script will likely abort. You can only use non-zero exit codes to signify specific errors, but successfully enabling the maintenance mode is not an error.

Reply:
Well, the exit code `1` is only returned if you _query_ for the status. And the most likely case if I query the status on deployment is that I _want_ it to fail if the mode is already enabled? Because for example I don't want to do another deployment that disables the maintenance mode?

Reply:
If you really need that we should add a parameter like `--fail-if-enabled` (that works in every combination of format etc.)

This way you can use something like this:

```sh
$ contao:maintenance-mode --fail-if-enabled && do-something-only-if-not-in-maintenance foo bar
```

Reply:
From working on the command line in linux I, would expect an error code of 0 when successfully querying information. We should IMHO conform with that. 

The [Advanced Bash-Scripting Guide](https://tldp.org/LDP/abs/html/exit-status.html) says:
> 	A successful command returns a 0, while an unsuccessful one returns a non-zero value that usually can be interpreted as an error code. Well-behaved UNIX commands, programs, and utilities return a 0 exit code upon successful completion, though there are some exceptions.

Reply:
I like the `--fail-if-enabled` if it is really necessary to return something other than `0` (which I still don't think it is).

Reply:
Changed in ba015aa126d342e9113844f56244316f20c9762b